src/services/user-config-service.ts

```diff
@@ -56,12 +56,26 @@ export const updateRepoConfig = async (
 /**
  * Returns the config for a given repo.
  */
-export const getRepoConfig = async (subscription: Subscription, repositoryId: number): Promise<Config | undefined> => {
+export const getRepoConfig = async (
+	subscription: Subscription,
+	installationId: InstallationId,
+	repositoryId: number,
+	repoOwner: string,
+	repoName: string
+): Promise<Config | undefined> => {
 	// In the future, we may look in other places for a config than just in the RepoSyncState (for example,
 	// we might fall back to default configs on the level of a subscription or an installation).
 	const repoSyncState = await RepoSyncState.findByRepoId(subscription, repositoryId);
-	// TODO: if repoSyncState is not available, try to get the config file from GitHub!
-	return repoSyncState?.config;
+
+	// Edge case: we don't have a record of the repository in our DB, yet, so we're loading the
+	// config directly from the config file in the GitHub repo.
+	if (!repoSyncState) {
+		const yamlConfig = await getRepoConfigFromGitHub(installationId, repoOwner, repoName);
+		return convertYamlToUserConfig(yamlConfig);
+	}
```

Comment:
Added handling of this edge case as it might be that we don't have a `RepoSyncState` record in the database, yet, for example, when a customer has many repos and the repo in question is last in line to be "discovered" at the start of a backfill. 

In this case, we fall back to loading the config directly from GitHub.